code/game/objects/structures/crates_lockers/crates/secure.dm

```diff
@@ -35,6 +35,10 @@
 	for(var/atom/movable/AM in src)
 		qdel(AM)
 	explosion(get_turf(src), 0, 1, 5, 5)
+	var/message = "[ADMIN_LOOKUPFLW(user)] has detonated [src.name]."

```

Comment:
The if_user check up above implies there isn't always a user when the crate goes boom, so you should move this logging to be under that same check.